### PR TITLE
Nickakhmetov/HMP-103 Improve publication cypress tests

### PR DESCRIPTION
**Pull request by NickAkhmetov**
- Improved existing tests by removing explicit `waits` and testing for the attribute affected by the action to be `true` instead
- Added more `data-testid`'s where needed
- Added tests for the rest of the publication page

Please note that I did this work off the linting branch to avoid potential conflicts.

image

### CODE REVIEW

File: CHANGELOG-lint-updates.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+- Update project ESLint configuration and dependencies.
```
Comment (john-conroy) [inline CHANGELOG-lint-updates.md:1]: This seems to have escaped the merge.

**Review by john-conroy — APPROVED** [review]
Looks good to me!

File: CHANGELOG-lint-updates.md (hunk 1 above)
Comment (NickAkhmetov) [reply CHANGELOG-lint-updates.md:1]: 👀 on it, I think this must have gotten propagated from an earlier merge

**Comment by NickAkhmetov** [comment]
- The old CHANGELOG has been removed
- I've pushed a fix for the cypress test case that failed - I'm not sure how it ever passed since Publication Provenance sections aren't supposed to have tables (per https://github.com/hubmapconsortium/portal-ui/blob/nickakhmetov/improve-publication-cypress-tests/context/app/static/js/components/detailPage/provenance/ProvTabs/ProvTabs.jsx#L21)

**Review by john-conroy — APPROVED** [review]

**Pull request merged**